### PR TITLE
Introduce UserAuthenticator and rename Authorizer to ChannelAuthorizer

### PR DESCRIPTION
**Pull request by agatav**
## Description

- rename `Authorizer` to `ChannelAuthorizer`
- rename `HttpAuthorizer` class to `HttpChannelAuthorizer`
- add `HttpUserAuthenticator`

Next PR: create user class, add signin(), add bindings for user & watchlist events.

### CODE REVIEW

File: PusherClient/PusherOptions.cs [hunk 1]
```diff
@@ -24,9 +24,9 @@ public PusherOptions()
         public bool Encrypted { get; set; }
 
         /// <summary>
-        /// Gets or set the <see cref="IAuthorizer"/> to use.
+        /// Gets or set the <see cref="IChannelAuthorizer"/> to use.
         /// </summary>
-        public IAuthorizer Authorizer { get; set; } = null;
```
Comment (sonologico) [inline PusherClient/PusherOptions.cs:29]: Should we keep both for backwards compatibility?

File: README.md [hunk 2]
```diff
@@ -101,7 +101,7 @@ AutoResetEvent doneEvent = new AutoResetEvent(false);
 // Create Pusher client ready to subscribe to public, private and presence channels
 Pusher pusher = new Pusher(Config.AppKey, new PusherOptions
 {
-    Authorizer = new FakeAuthoriser(),
```
Comment (sonologico) [inline README.md:104]: The name of the field also needs to updated

File: PusherClient/PusherOptions.cs (hunk 1 above)
Comment (fbenevides) [reply PusherClient/PusherOptions.cs:29]: I guess so. Maybe adding some Deprecated comment or something

Comment (agatav) [reply PusherClient/PusherOptions.cs:29]: done

**Pull request merged**